tests/repository/test_repository.py

```diff
@@ -566,6 +581,37 @@ def test_walk(repository, generate_directory):
     ]
 
 
+def test_copy_tree(repository, generate_directory, tmp_path_parametrized):
+    """Test the ``Repository.copy_tree`` method."""
+    directory = generate_directory({'file_a': None, 'relative': {'file_b': None, 'sub': {'file_c': None}}})
+    repository.put_object_from_tree(str(directory))
+
+    repository.copy_tree(tmp_path_parametrized)
+    for root, dirnames, filenames in repository.walk():
+        for dirname in dirnames:
+            assert pathlib.Path(tmp_path_parametrized / root / dirname).is_dir()
+        for filename in filenames:
+            filepath = pathlib.Path(tmp_path_parametrized / root / filename)
+            assert filepath.is_file()
+            with repository.open(root / filename) as handle:
+                assert filepath.read_bytes() == handle.read()
+
+
+@pytest.mark.parametrize(('argument', 'value', 'exception'), (
+    ('target', None, 'path is not of type `str` nor `pathlib.Path`.'),
+    ('target', 'relative/path', 'provided target is not an absolute path.'),
+    ('target', pathlib.Path('.'), 'provided target is not an absolute path.'),
+))
+def test_copy_tree_invalid(tmp_path, repository, argument, value, exception):
+    """Test the ``Repository.copy_tree`` method for invalid input."""
+    if argument == 'target':
+        with pytest.raises(TypeError, match=exception):
+            repository.copy_tree(target=value)
+    else:
+        with pytest.raises(TypeError, match=exception):
+            repository.copy_tree(target=tmp_path, path=value)
+
```

Comment:
Why do you have this `if`, all your parameters have `argument == 'target'`, right? Was this supposed to support also testing the errors on `path`? (although I see no raises related to that variable...should we perhaps raise if the `path` doesn't exist in the node or would the raise inside `Repository.open` give enough info to understand the problem?)

Reply:
Yeah, somehow one of the test cases where `path` is specified got lost. I will add it.

Reply:
I see it now, thanks! However there is no added check for the path existing in the node repo no? 😅 

I mean perhaps it is not necessary and the error that `self.walk(path)` would raise should be enough for the user to identify where he is making the error (if it already mentions the path and the node id, for example), but I just wanted explicit confirmation that it is ok to rely on that here.

Reply:
> However there is no added check for the path existing in the node repo no?

There is though. Granted, the check is led from the perspective of the source `repository`, but it uses `walk(path=path)` to check that all the files that should have been copied, actually exist in `tmp_path_parametrized`. Of course, if `walk` is bugged, we could have a false positive in some really weird obscure case, potentially. But that method itself is also tested so I think it is fine.

Reply:
Rereading your last comment, I think you may have put this under the wrong initial comment which concerns the test for invalid arguments. That does not include the use of `walk` whatsoever. Probably you meant to say this with the other comment below on the `test_copy_tree` test, which does include the use of `walk`. My answer applies the same though.

Reply:
> Granted, the check is led from the perspective of the source `repository`, but it uses `walk(path=path)` to check that all the files that should have been copied

Right, ok, so this is what I meant with "relying on `self.walk(path)` to do the check and raise". It is just that sometimes lower level functions give less informative errors and you need to `try / except` and re-raise with more info, but I guess here it should be fine.